pkg/controller/controlplane/valuesprovider.go

```diff
@@ -477,7 +477,13 @@ func getCSIControllerChartValues(
 		},
 	}
 
-	if semver.Compare(cluster.Shoot.Spec.Kubernetes.Version, "1.30.0") > 0 {
+	// golang mod's semver considers only semvers starting with 'v' to be proper semver, so we need to do some checking/converting here.
+	// On the bright side, it can handle the shoot-spec-supported case of partial semvers, e.g.: (v)1.31
+	k8sVersion := cluster.Shoot.Spec.Kubernetes.Version
+	if !strings.HasPrefix(k8sVersion, "v") {
+		k8sVersion = "v" + k8sVersion
+	}
+	if semver.Compare(k8sVersion, "v1.30.0") > 0 {
```

Comment:
Instead of the hack we maybe could make use `versionutils.ConstraintK8sLess130.Check(k8sVersion)` like in this example: https://github.com/gardener/gardener-extension-provider-gcp/blob/07acf111fa7321939d63be6c36597cdacde0895c/pkg/webhook/controlplane/ensurer.go#L183

Reply:
👍 on the suggestion